src/completion-provider/script-completion-provider.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  CompletionItem,
+  TextDocumentPositionParams
+} from 'vscode-languageserver';
+
+import Server from '../server';
+import ASTPath from '../glimmer-utils';
+import { toPosition } from '../estree-utils';
+import { filter } from 'fuzzaldrin';
+import { parseScriptFile as parse } from 'ember-meta-explorer';
+const { uniqBy } = require('lodash');
+const memoize = require('memoizee');
+import { getExtension } from '../utils/file-extension';
+import { log } from '../utils/logger';
+import {
+  isStoreModelLookup,
+  isRouteLookup,
+  isModelReference,
+  isNamedServiceInjection,
+  isTransformReference,
+  isComputedPropertyArgument
+} from '../utils/ast-helpers';
+import {
+  listRoutes,
+  listModels,
+  listServices,
+  mGetProjectAddonsInfo,
+  listTransforms
+} from '../utils/layout-helpers';
+
+const mListRoutes = memoize(listRoutes, { length: 1, maxAge: 60000 });
+const mListModels = memoize(listModels, { length: 1, maxAge: 60000 });
+const mListServices = memoize(listServices, { length: 1, maxAge: 60000 });
+const mListTransforms = memoize(listTransforms, { length: 1, maxAge: 60000 });
+
+export default class ScriptCompletionProvider {
+  constructor(private server: Server) {}
+  provideCompletions(params: TextDocumentPositionParams): CompletionItem[] {
+    log('provideCompletions');
+    if (!['.js', '.ts'].includes(getExtension(params.textDocument) as string)) {
```

Comment:
The fact that this requires a cast suggests a smell here! I dug in and discovered that `getExtension` doesn't actually need to return `| null` at all, so was able to clean this up.